apprentice_teacher.rb

```diff
@@ -1,50 +1,17 @@
+require_relative 'dbc_module'
+require_relative 'student'
+require_relative 'teacher_module'
+require_relative 'senior_teacher'
+
 class ApprenticeTeacher
-  attr_reader :age, :salary, :phase, :target_raise
-  attr_accessor :name
+  include DbcMod
+  include TeacherMod
+  attr_reader :salary, :target_raise
 
   def initialize(options={})
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
-    @target_raise = 800
+    super(options)
     @phase = 3
-  end
```

Comment:
In object oriented programming, we tend to prefer using inheritance over modules. This way will certainly work, but we generally only use modules for things that aren't central to the object they live in, or which don't make sense in an existing superclass (and it doesn't make sense to add in a middle layer superclass). In this case, this behavior is very central to what it means to be a teacher at dbc, so they should be part of the superclass for apprentice and senior teacher, not a separate module.